src/data/DataStore.ts

```diff
@@ -408,9 +409,13 @@ class DataStore {
                 for (let dimIdx = 0; dimIdx < dimLen; dimIdx++) {
                     const dimStorage = chunks[dimIdx];
                     // PENDING NULL is empty or zero
-                    const val = this._dimValueGetter(
+                    let val = this._dimValueGetter(
                         dataItem, dimNames[dimIdx], idx, dimIdx
                     ) as ParsedValueNumeric;
+
+                    if (this._dimensions[dimIdx].type === 'time') {
```

Comment:
It's a very hotspot code so performance should be taken carefully. And I'm not sure but this fix seems more like a workaround. One thing that can be sure is that this bug is brought in https://github.com/apache/echarts/pull/15355. I will figure out why `time` format data breaks in this case.

Reply:
```js
const a = new Float64Array();
a[0] = '2018-01-02';
console.log(a) // [NaN]
```

Reply:
By the original design. The string time format will be stored in a normal array. Otherwise, it will be parsed in dimValueGetter like in https://github.com/apache/echarts/blob/53fa03d271e1ca696ebec6ae2c94e76e946883ae/src/data/helper/dataValueHelper.ts#L58

But I suppose this is broken after some changes.